src/main/java/in/org/projecteka/hiu/consent/ConceptValidator.java

```diff
@@ -0,0 +1,110 @@
+package in.org.projecteka.hiu.consent;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.InitializingBean;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.core.io.Resource;
+import org.springframework.stereotype.Component;
+import reactor.core.publisher.Mono;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+@Component
+public class ConceptValidator implements InitializingBean, ConceptLookup {
+    private static final Logger logger = LoggerFactory.getLogger(ConceptValidator.class);
+
+    @Value("${hiu.valueSets}")
+    private Resource valueSetsResource;
+    private Map<String, String> purposesOfUse;
+    private Map<String, String> hiTypes;
+
+    @Override
+    public void afterPropertiesSet() throws Exception {
+        ConcurrentHashMap<String, Map<String, String>> vsCodeMap = readValueSetFromResource();
+        purposesOfUse = vsCodeMap.get(HEALTH_INFORMATION_PURPOSE_OF_USE);
+        hiTypes = vsCodeMap.get(HEALTH_INFORMATION_TYPE);
+    }
+
+
+    private ConcurrentHashMap<String, Map<String, String>> readValueSetFromResource() throws IOException {
+        try (InputStream in = valueSetsResource.getInputStream()) {
+            ByteArrayOutputStream result = new ByteArrayOutputStream();
+            byte[] buffer = new byte[1024];
```

Comment:
Would not it be awesome to make it more obvious by defining class with properties which are required to read value set 🤔 and easier for new person who's going to onboard to understand?

Reply:
Except this and some extra empty spaces it's looking good.

Reply:
as in different valuesets for purpose and HIType? Thought about it, but then for simplicity sake, kept in a single json. UI also uses the same.

Reply:
Maybe you meant different validators for different concepts? Thought about that too. Then thought these would be the same. Conceptually, they are just validating same type of concepts, just from different system/resource perspective.